### PR TITLE
TW-440: Fix DataURI render in Collectibles' icons

### PR DESCRIPTION
**Pull request by alex-tsx**

### CODE REVIEW

File: src/components/token-icon/token-icon.tsx [hunk 1]
```diff
@@ -41,20 +43,30 @@ export const TokenIcon: FC<Props> = ({ iconName, thumbnailUri, size = formatSize
 
   return (
     <View style={[TokenIconStyles.container, { borderRadius: size / 2 }]}>
-      {isDefined(iconName) ? (
-        <Icon name={iconName} color={metadata.iconName === iconName ? colors.black : undefined} size={size} />
-      ) : isString(thumbnailUri) ? (
-        isImgUriSvg(thumbnailUri) ? (
-          <SvgCssUri width={size} height={size} uri={thumbnailUri} />
-        ) : (
+      {(() => {
```
Comment (herkoss) [inline src/components/token-icon/token-icon.tsx:46]: Can we move this logic into another component?

File: src/store/tokens-metadata/tokens-metadata-epics.ts [hunk 2]
```diff
@@ -37,7 +37,11 @@ const loadTokenSuggestionEpic = (action$: Observable<Action>) =>
           loadTokenSuggestionActions.success(tokenMetadata),
           addTokensMetadataAction([tokenMetadata])
         ]),
-        catchError(err => of(loadTokenSuggestionActions.fail(err.message)))
+        catchError(error => {
+          console.error(error);
```
Comment (herkoss) [inline src/store/tokens-metadata/tokens-metadata-epics.ts:41]: Can we remove this console.log?

File: src/utils/token-metadata.utils.ts [hunk 3]
```diff
@@ -147,11 +147,28 @@ export const loadWhitelist$ = (selectedRpc: string): Observable<Array<TokenMetad
...
-  (address: string, id = 0): Observable<TokenMetadataInterface> =>
-    from(tezosMetadataApi.get<TokenMetadataResponse>(`/metadata/${address}/${id}`)).pipe(
+  (address: string, id = 0): Observable<TokenMetadataInterface> => {
+    const slug = `${address}_${id}`;
+    console.log('Loading metadata for:', slug);
+
+    return from(
+      tezosMetadataApi.get<TokenMetadataResponse>(`/metadata/${address}/${id}`).then(
+        res => {
+          console.log('Metadata load result:', res.data);
```
Comment (herkoss) [inline src/utils/token-metadata.utils.ts:157]: We need to remove these console.logs

**Review by herkoss — COMMENTED** [review]
Great!)

File: src/utils/token-balance.utils.ts [hunk 4]
```diff
@@ -1,40 +1,32 @@
...
+import { getTokenSlug, getTokenType } from 'src/token/utils/token.utils';
+
 import { isDefined } from './is-defined';
 import { readOnlySignerAccount } from './read-only.signer.util';
 import { createReadOnlyTezosToolkit } from './rpc/tezos-toolkit.utils';
 
 const TEZOS_DOMAINS_NAME_REGISTRY_ADDRESS = 'KT1GBZmSxmnKJXGMdMLbugPfLyUPmuLSMwKS';
 
-const limit = 300;
+const LIMIT = 600;
```
Comment (unixvb) [inline src/utils/token-balance.utils.ts:16]: check please if big limit affects request performance.  
tz1MnoSe4HMAzfoyiPGkrAL6a3WzBhRQRba9

**Review by unixvb — APPROVED** [review]

**Review by unixvb — DISMISSED** [review]

**Review by unixvb — APPROVED** [review]

**Review by IrynaKhyzhynska — APPROVED** [review]

**Review by NovakVlad2424 — APPROVED** [review]
Done IOS: 1.13_557

**Pull request merged**